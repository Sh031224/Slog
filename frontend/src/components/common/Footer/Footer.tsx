import React from "react";
import Link from "next/link";
import "./Footer.scss";

const Footer = () => {
  return (
    <footer className="footer">
      <div className="footer-container">
        <a href="https://github.com/Sh031224" target="_blank">
          © 2020. Sh031224. All rights reserved.
        </a>
        <Link href="/privacy">
          <a className="footer-container-privacy">개인정보 처리방침</a>
        </Link>
      </div>
    </footer>
  );
};

export default React.memo(Footer);
