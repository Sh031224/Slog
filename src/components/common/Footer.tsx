import Link from "next/link";
import React from "react";
import styled from "styled-components";

const Footer: React.FC = () => {
  return (
    <FooterWrapper>
      <FooterContainer>
        <a href="https://github.com/Sh031224" target="_blank" rel="noopener">
          © 2020. Sh031224. All rights reserved.
        </a>
        <>
          <Link href="/privacy">
            <a>
              <FooterPrivacy>개인정보 처리방침</FooterPrivacy>
            </a>
          </Link>
        </>
      </FooterContainer>
    </FooterWrapper>
  );
};

const FooterWrapper = styled.footer`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2rem;
`;

const FooterContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  ${({ theme }) => theme.device.mobile} {
    flex-direction: column;
  }
  & > a {
    text-decoration: none;
    color: ${({ theme }) => theme.colors.ftGray};
    font-size: 0.9rem;
  }
`;

const FooterPrivacy = styled.div`
  color: ${({ theme }) => theme.colors.ftDarkGray} !important;
  margin-left: 1rem;
  display: inline-block;
  ${({ theme }) => theme.device.mobile} {
    margin-left: 0;
  }
`;

export default Footer;
