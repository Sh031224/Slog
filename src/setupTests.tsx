import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { lightTheme } from "style/theme";
import { store } from "stores";
import { Provider } from "react-redux";
import { NotificationContainer } from "react-notifications";

const Providers = ({ children }) => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={lightTheme}>
        <NotificationContainer />
        {children}
      </ThemeProvider>
    </Provider>
  );
};

const customRender = (ui, options = {}) => render(ui, { wrapper: Providers, ...options });

// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

(window as any).scrollTo = () => {};

export { customRender as render };
