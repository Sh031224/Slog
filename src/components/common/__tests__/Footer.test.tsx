import { render } from "setupTests";
import Footer from "../Footer";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: ""
    };
  }
}));

describe("Footer", () => {
  const setup = (props = {}) => {
    return render(<Footer {...props} />);
  };

  it("renders correctly", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
