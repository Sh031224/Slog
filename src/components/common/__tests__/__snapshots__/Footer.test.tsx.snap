// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #777;
  font-size: 0.9rem;
}

.c2 {
  color: #464646 !important;
  margin-left: 1rem;
  display: inline-block;
}

@media only screen and (max-width:500px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media only screen and (max-width:500px) {
  .c2 {
    margin-left: 0;
  }
}

<div>
  <div
    class="notification-container notification-container-empty"
  >
    <div />
  </div>
  <footer
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        href="https://github.com/Sh031224"
        rel="noopener"
        target="_blank"
      >
        © 2020. Sh031224. All rights reserved.
      </a>
      <a
        href="/privacy"
      >
        <div
          class="c2"
        >
          개인정보 처리방침
        </div>
      </a>
    </div>
  </footer>
</div>
`;
